### PR TITLE
Add `gcc`, `gfortran` and `clang` on Windows and Ubuntu images

### PR DESCRIPTION
**Pull request by ivan-nosar**
This PR adds a bunch of software tools on VM images:
- `gcc`
  - Ubuntu: `gcc-7` was installed on image previously. `gcc-8` and `gcc-9` added.
  - Windows: `gcc-8` was installed on image previously in scope of `MinGW` tool. No new version added.
- `gfortran`
  - Ubuntu: `gfortran-8` and `gfortran-9` added
  - Windows: `gfortran-8` was installed on image previously in scope of `MinGW` tool. No new version added.
- `Clang`
  - Ubuntu: `clang-6` was installed on image previously. `clang-8` and `clang-9` was added (along with `llvm-8` and `llvm-9`)
  - Windows: `clang-9` was added along with `llvm-9`

### CODE REVIEW

File: images/linux/scripts/installers/clang.sh [hunk 1]
```diff
@@ -1,31 +1,77 @@
...
+    for cmd in clang-6.0 clang++-6.0; do
+        if ! command -v $cmd; then
+            echo "$cmd was not installed"
+            exit 1
+        fi
+    done
+
+    # Document what was added to the image
+    echo "Lastly, documenting what we added to the metadata file"
+    DocumentInstalledItem "Clang 6.0 ($(clang-6.0 --version | head -n 1 | cut -d ' ' -f 3 | cut -d '-' -f 1))"
```
Comment (al-cheb) [inline images/linux/scripts/installers/clang.sh]: This function fully duplicates InstallClang.

File: images/win/scripts/Installers/Validate-Clang.ps1 [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+# Adding description of the software to Markdown
+
+# `clang --version` gives output like:
+# clang version 9.0.0 (tags/RELEASE_900/final)
+# Target: x86_64-pc-windows-msvc
+# Thread model: posix
+# InstalledDir: C:\Program Files\LLVM\bin
+
+$SoftwareName = "Clang"
+$(clang --version).Split([System.Environment]::NewLine)[0] -match "\d\.\d\.\d"
```
Comment (al-cheb) [inline images/win/scripts/Installers/Validate-Clang.ps1]: If clang were updated to xx.xx.xx?

File: images/linux/scripts/installers/clang.sh (hunk 1 above)
Comment (ivan-nosar) [reply images/linux/scripts/installers/clang.sh]: Fixed, thanks

File: images/win/scripts/Installers/Validate-Clang.ps1 (hunk 2 above)
Comment (ivan-nosar) [reply images/win/scripts/Installers/Validate-Clang.ps1]: Fixed, thanks

File: images/linux/scripts/installers/clang.sh [hunk 3]
```diff
@@ -1,31 +1,62 @@
...
+# Download script for automatic installation
+wget https://apt.llvm.org/llvm.sh
+chmod +x llvm.sh
+
+# Install Clang 8 and 9
+versions=(
+    "6.0"
+    "8"
+    "9"
+)
```
Comment (sergey-akhalkov) [inline images/linux/scripts/installers/clang.sh]: nitpick: looks weird a bit, comment is about 8 and 9 but below I can see 6, 8 and 9

Comment (ivan-nosar) [reply images/linux/scripts/installers/clang.sh]: Fixed, thanks

**Comment by q4a** [comment]
Why did you remove clang-9 and llvm-9 from Windows?
I'm talking about this commit:
https://github.com/actions/virtual-environments/pull/369/commits/9c06dae2e32a6f9fefa528aa5b764237ca293626

**Comment by EwoutH** [comment]
@ivan-nosar Could you add GCC 10 to the Ubuntu 18.04 image? It's available on [Debian testing](https://tracker.debian.org/pkg/gcc-10) so it can be installed with apt.

**Comment by maxim-lobanov** [comment]
@EwoutH , feel free to create the issue for that.

**Comment by autodidaddict** [comment]
> Why did you remove clang-9 and llvm-9 from Windows?
> I'm talking about this commit:
> [9c06dae](https://github.com/actions/virtual-environments/commit/9c06dae2e32a6f9fefa528aa5b764237ca293626)

Came here to find out the same thing. I've got a windows build failing because clang is missing.

**Comment by advancedwebdeveloper** [comment]
Is it possible to provide https://github.com/llvm/llvm-project/releases/tag/llvmorg-11.0.0 ?
Clang 11, lld and libc++ of the latest stable version, I mean.

**Comment by miketimofeev** [comment]
@advancedwebdeveloper could you please create an issue with this request?

**Pull request merged**